src/Symfony/Component/Form/Extension/Core/DataTransformer/StringToFloatTransformer.php

```diff
@@ -34,7 +34,7 @@ public function transform($value)
             return null;
         }
 
-        if (!\is_string($value) || !is_numeric($value)) {
+        if ((!\is_string($value) || !is_numeric($value)) && !\is_float($value)) {
             throw new TransformationFailedException('Expected a numeric string.');
```

Comment:
Maybe the message can also be improved: `'Expected a float or a numeric string.'`?